src/budget/budgetmanager.cpp

```diff
@@ -1327,15 +1327,46 @@ void CBudgetManager::SetSynced(bool synced)
     }
 }
 
+template<typename Map, typename Item>
+static bool relayItemIfFound(const uint256& itemHash, CNode* pfrom, RecursiveMutex& cs, Map& map, bool fPartial, const char* type)
```

Comment:
Since this is always going to be a map hash->item, you could simplify the code a bit by using a single templated argument
```suggestion
template<typename T>
static bool relayItemIfFound(const uint256& itemHash, CNode* pfrom, RecursiveMutex& cs, std::map<uint256, T>& map, bool fPartial, const char* type)
```

Reply:
sounds good